ts/Stock/Indicators/RSI/RSIIndicator.ts

```diff
@@ -106,20 +107,28 @@ class RSIIndicator extends SMAIndicator {
             change: number,
             avgGain: number,
             avgLoss: number,
-            i: number;
+            i: number,
+            values: Array<number>;
 
         // RSI requires close value
-        if (
-            (xVal.length < period) || !isArray(yVal[0]) ||
-            yVal[0].length !== 4
-        ) {
+        if ((xVal.length < period)) {
             return;
         }
+        if (!isArray(yVal[0])) {
+            values = yVal as Array<number>;
+        } else if (yVal[0].length < 4) {
+            values = (yVal as Array<Array<number>>).map(
+                (value: Array<number>): number =>
+                    value[(params.index as number) || (yVal as Array<Array<number>>)[0].length - 1]
+            );
```

Comment:
I'm not sure what is the purpose of this block. Is this support for series types that have less than four values in data (e.g. arearange)? Could we instead check this way: 

```js
if (!isArray()) {
 ...
} else {
  let index = Math.min(params.index, yVal[0].length - 1);
  values = (yVal as any).map((value: Array<number>): number => value[index]);
}
```